src/parser.c

```diff
@@ -40,6 +46,23 @@ void parse_array_init(var_t *var,
                       basic_block_t **bb,
                       bool emit_code);
 
+
+label_t *find_label(char *name)
+{
+    for (int i = 0; i < label_idx; i++) {
+        if (!strcmp(name, labels[i].label_name))
+            return &labels[i];
+    }
+    return NULL;
+}
+
+void add_label(char *name, basic_block_t *bb)
+{
+    label_t *l = &labels[label_idx++];
```

Comment:
We need to check here if the number of labels exceeds MAX_LABELS and report an error if so, to prevent potential out-of-bounds access.